op-node/rollup/derive/channel_bank.go

```diff
@@ -38,18 +38,20 @@ type ChannelBank struct {
 	progress Progress
 
 	next ChannelBankOutput
+	prev *L1Retrieval
 }
 
 var _ Stage = (*ChannelBank)(nil)
 
 // NewChannelBank creates a ChannelBank, which should be Reset(origin) before use.
-func NewChannelBank(log log.Logger, cfg *rollup.Config, next ChannelBankOutput) *ChannelBank {
+func NewChannelBank(log log.Logger, cfg *rollup.Config, next ChannelBankOutput, prev *L1Retrieval) *ChannelBank {
```

Comment:
I think it should be possible for each stage to instantiate its predecessor directly, no need to pass it in, or to coordinate in `pipeline.go` at all.